src/monodroid/jni/embedded-assemblies.hh

```diff
@@ -81,6 +83,12 @@ namespace xamarin::android::internal {
 #endif // def NET6
 		MonoReflectionType* typemap_java_to_managed (MonoString *java_type);
 
+		void prepare_for_multiple_threads () noexcept
+		{
+			int ret = sem_init (&assembly_mmap_semaphore, 0 /* pshared */, 1 /* value */);
+			abort_unless (ret == 0, "Failed to initialize assembly mapping semaphore. %s", strerror (errno));
```

Comment:
I think this `abort` is happening during the Designer tests since this got merged?

I got this from their log, is it useful?
```
[2021-08-18 23:21:55.5] Renderer >> =================================================================
[2021-08-18 23:21:55.5] Renderer >> 	Native Crash Reporting
[2021-08-18 23:21:55.5] Renderer >> =================================================================
[2021-08-18 23:21:55.5] Renderer >> Got a abrt while executing native code. This usually indicates
[2021-08-18 23:21:55.5] Renderer >> a fatal error in the mono runtime or one of the native libraries 
[2021-08-18 23:21:55.5] Renderer >> used by your application.
[2021-08-18 23:21:55.5] Renderer >> =================================================================
[2021-08-18 23:21:55.5] Renderer >> 
[2021-08-18 23:21:55.5] Renderer >> =================================================================
[2021-08-18 23:21:55.5] Renderer >> 	Native stacktrace:
[2021-08-18 23:21:55.5] Renderer >> =================================================================
[2021-08-18 23:21:55.5] Renderer >> 	0x11d9349d8 - /Library/Frameworks/Xamarin.Android.framework/Versions/11.4.99.110/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-Darwin/libmonosgen-2.0.dylib : mono_dump_native_crash_info
[2021-08-18 23:21:55.5] Renderer >> 	0x11d8cf38e - /Library/Frameworks/Xamarin.Android.framework/Versions/11.4.99.110/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-Darwin/libmonosgen-2.0.dylib : mono_handle_native_crash
[2021-08-18 23:21:55.5] Renderer >> 	0x11d933fdf - /Library/Frameworks/Xamarin.Android.framework/Versions/11.4.99.110/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-Darwin/libmonosgen-2.0.dylib : sigabrt_signal_handler
[2021-08-18 23:21:55.5] Renderer >> 	0x7fff2043dd7d - /usr/lib/system/libsystem_platform.dylib : _sigtramp
[2021-08-18 23:21:55.5] Renderer >> 	0x1060aa468 - Unknown
[2021-08-18 23:21:55.5] Renderer >> 	0x7fff2034d406 - /usr/lib/system/libsystem_c.dylib : abort
[2021-08-18 23:21:55.5] Renderer >> 	0x11d7f0cbd - /Library/Frameworks/Xamarin.Android.framework/Versions/11.4.99.110/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-Darwin/libmono-android.debug.dylib : _ZL15do_abort_unlessbPKcz
[2021-08-18 23:21:55.5] Renderer >> 	0x11d7eda7a - /Library/Frameworks/Xamarin.Android.framework/Versions/11.4.99.110/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-Darwin/libmono-android.debug.dylib : _ZN7xamarin7android8internal16MonodroidRuntime38Java_mono_android_Runtime_initInternalEP7JNIEnv_P7_jclassP8_jstringP13_jobjectArrayS8_SA_P8_jobjectSA_ih
[2021-08-18 23:21:55.5] Renderer >> 	0x11d7ed005 - /Library/Frameworks/Xamarin.Android.framework/Versions/11.4.99.110/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-Darwin/libmono-android.debug.dylib : Java_mono_android_Runtime_init
[2021-08-18 23:21:55.5] Renderer >> 	0x1061b33a7 - Unknown
[2021-08-18 23:21:55.5] Renderer >> 	0x1061a2ffd - Unknown
[2021-08-18 23:21:55.5] Renderer >> 	0x1061a27d0 - Unknown
[2021-08-18 23:21:55.5] Renderer >> 
[2021-08-18 23:21:55.5] Renderer >> =================================================================
[2021-08-18 23:21:55.5] Renderer >> 	Basic Fault Address Reporting
[2021-08-18 23:21:55.5] Renderer >> =================================================================
[2021-08-18 23:21:55.5] Renderer >> Memory around native instruction pointer (0x7fff203c992e):0x7fff203c991e  ff ff c3 90 90 90 b8 48 01 00 02 49 89 ca 0f 05  .......H...I....
[2021-08-18 23:21:55.5] Renderer >> 0x7fff203c992e  73 08 48 89 c7 e9 a1 a1 ff ff c3 90 90 90 b8 53  s.H............S
[2021-08-18 23:21:55.5] Renderer >> 0x7fff203c993e  00 00 02 49 89 ca 0f 05 73 08 48 89 c7 e9 89 a1  ...I....s.H.....
[2021-08-18 23:21:55.5] Renderer >> 0x7fff203c994e  ff ff c3 90 90 90 b8 83 01 00 02 49 89 ca 0f 05  ...........I....
[2021-08-18 23:21:55.5] Renderer >> 
[2021-08-18 23:21:55.5] Renderer >> =================================================================
[2021-08-18 23:21:55.5] Renderer >> 	Managed Stacktrace:
[2021-08-18 23:21:55.5] Renderer >> =================================================================
[2021-08-18 23:21:55.5] Renderer >> =================================================================
```

Reply:
It does help, thanks!